jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaHandlerResolver.kt

```diff
@@ -39,9 +39,20 @@ class JavaLambdaHandlerResolver : LambdaHandlerResolver {
                     classes.filterIsInstance<NavigatablePsiElement>().toTypedArray()
                 } else {
                     val handlerMethod = classes.asSequence()
-                        .map { it.findMethodsByName(methodName, true) }
+                        .map { psiClass ->
+                            psiClass.findMethodsByName(methodName, true)
+                                .filter { it.body != null } // Filter out interfaces
+                                .filter {
+                                    val file = it.containingFile.virtualFile
+
+                                    return@filter if (psiClass.implementsLambdaHandlerInterface(file)) {
+                                        true
+                                    } else {
+                                        it.isValidHandler(psiClass, file)
+                                    }
+                                }
+                        }
```

Comment:
Should the order of the two filters be reversed?

Reply:
the second filter is a more expensive check, so cheaper one is first